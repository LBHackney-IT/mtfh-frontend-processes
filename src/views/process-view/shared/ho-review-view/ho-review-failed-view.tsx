import { locale } from "../../../../services";
import { IProcess } from "../../../../types";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { StatusBox, Text } from "@mtfh/common/lib/components";

const { views } = locale;
interface HoReviewFailedViewProps {
  processConfig: IProcess;
  process: Process;
}

export const HoReviewFailedView = ({
  processConfig,
  process,
}: HoReviewFailedViewProps): JSX.Element => {
  const { states } = processConfig;
  const { hoApprovalFailed } = states;
  const { currentState, previousStates } = process;
  const hoApprovalFailedState =
    currentState.state === hoApprovalFailed.state
      ? currentState
      : previousStates.find((previous) => previous.state === hoApprovalFailed.state);

  return (
    <>
      <StatusBox
        variant="warning"
        title={views.hoReviewView.hoOutcome(
          views.hoReviewModal[process.processName.toLowerCase()],
          "declined",
        )}
      >
        {hoApprovalFailedState?.processData.formData.reason && (
          <Text>{hoApprovalFailedState.processData.formData.reason}</Text>
        )}
      </StatusBox>
    </>
  );
};
