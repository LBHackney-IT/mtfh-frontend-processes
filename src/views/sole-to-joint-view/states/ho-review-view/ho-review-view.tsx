import React, { useEffect, useState } from "react";

import { isPast } from "date-fns";
import { Form, Formik } from "formik";

import { AppointmentDetails } from "../../../../components/appointment-details/appointment-details";
import {
  DateTimeFields,
  validate,
} from "../../../../components/appointment-form/appointment-form";
import { locale } from "../../../../services";
import { Trigger } from "../../../../services/processes/types";
import { IProcess } from "../../../../types";
import { TenantContactDetails, getAppointmentDateTime } from "../shared";

import { Process, editProcess } from "@mtfh/common/lib/api/process/v1";
import {
  Button,
  Checkbox,
  Field,
  Heading,
  InlineField,
  Input,
  Radio,
  RadioGroup,
  Text,
} from "@mtfh/common/lib/components";

interface HoReviewViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  setGlobalError: any;
  optional?: any;
}

enum Choice {
  Appointment = "appointment",
  Review = "review",
}

enum Decision {
  Approve = "approve",
  Decline = "decline",
}

export const HoReviewView = ({
  processConfig,
  process,
  mutate,
  setGlobalError,
  optional,
}: HoReviewViewProps): JSX.Element => {
  const [choice, setChoice] = useState<Choice | undefined>();
  const [disabled, setDisabled] = useState<boolean>(true);
  const [needAppointment, setNeedAppointment] = useState<boolean>(false);

  const { currentState } = process;
  const { interviewScheduled, interviewRescheduled } = processConfig.states;
  const formData = process.currentState.processData.formData as {
    appointmentDateTime: string;
  };

  useEffect(() => {
    if (needAppointment) {
      setChoice(Choice.Appointment);
    }
  }, [needAppointment]);

  return (
    <>
      {[interviewScheduled.state, interviewRescheduled.state].includes(
        process.currentState.state,
      ) && (
        <AppointmentDetails
          process={process}
          needAppointment={needAppointment}
          setNeedAppointment={setNeedAppointment}
          options={{
            requestAppointmentTrigger: Trigger.ScheduleInterview,
            rescheduleAppointmentTrigger: Trigger.RescheduleInterview,
            appointmentRequestedState: interviewScheduled.state,
            appointmentRescheduledState: interviewRescheduled.state,
          }}
        />
      )}

      <Formik
        initialValues={{
          decision: undefined,
          confirm: false,
          managerName: "",
          choice,
          day: "",
          month: "",
          year: "",
          hour: "",
          minute: "",
          amPm: "",
        }}
        enableReinitialize
        onSubmit={async (values) => {
          if (choice === Choice.Appointment) {
            const appointmentDateTime = getAppointmentDateTime(values);
            let processTrigger: Trigger | string = Trigger.ScheduleInterview;
            if (
              [interviewScheduled.state, interviewRescheduled.state].includes(
                currentState.state,
              )
            ) {
              if (isPast(new Date(formData.appointmentDateTime))) {
                processTrigger = Trigger.RescheduleInterview;
              } else {
                processTrigger = "";
              }
            }
            try {
              await editProcess({
                id: process.id,
                processTrigger,
                processName: process?.processName,
                etag: process.etag || "",
                formData: {
                  appointmentDateTime,
                },
                documents: [],
                processData: {
                  formData: {
                    appointmentDateTime,
                  },
                  documents: [],
                },
              });
              setNeedAppointment(false);
              setChoice(undefined);
              mutate();
            } catch (e: any) {
              setGlobalError(e.response?.status || 500);
            }
            return;
          }
          try {
            await editProcess({
              id: process.id,
              processTrigger: Trigger.HOApproval,
              processName: process?.processName,
              etag: process.etag || "",
              formData: {
                hORecommendation: values.decision,
              },
              documents: [],
            });
            mutate();
          } catch (e: any) {
            setGlobalError(e.response?.status || 500);
          }
        }}
        validateOnChange
        validateOnBlur
        validate={(values) => {
          console.log("validate", values.choice);
          if (values.choice === Choice.Appointment) {
            validate(values, setDisabled);
            return;
          }
          if (values.choice === Choice.Review) {
            if (
              values.decision !== undefined &&
              values.confirm &&
              values.managerName !== ""
            ) {
              setDisabled(false);
            } else {
              setDisabled(true);
            }
          }
        }}
      >
        {() => (
          <Form>
            <Field id="choice" name="choice" label="" type="radio">
              <RadioGroup>
                <Heading variant="h4" style={{ marginBottom: 26 }}>
                  Next steps
                </Heading>
                <Radio
                  id="ho-appointment"
                  value={Choice.Appointment}
                  onChange={() => {
                    setNeedAppointment(true);
                    setChoice(Choice.Appointment);
                  }}
                  checked={choice === Choice.Appointment && needAppointment}
                >
                  Make an appointment with the applicant for an interview
                </Radio>

                {choice === Choice.Appointment && needAppointment && (
                  <div style={{ marginLeft: 50, marginBottom: 20 }}>
                    <Text>
                      To make an appointment with the applicant for an interview, please
                      use the following details:
                    </Text>
                    {optional.tenant ? (
                      <>
                        <TenantContactDetails tenant={optional.tenant} />
                      </>
                    ) : (
                      <Text>Tenant not found.</Text>
                    )}
                    <DateTimeFields
                      dateLabel="Interview Date"
                      timeLabel="Interview Time"
                    />
                  </div>
                )}

                <Radio
                  id="ho-review"
                  value={Choice.Review}
                  onChange={() => {
                    setNeedAppointment(false);
                    setChoice(Choice.Review);
                  }}
                  checked={choice === Choice.Review}
                >
                  I have passed the case to the Housing Manager for review and received a
                  decision
                </Radio>

                {choice === Choice.Review && !needAppointment && (
                  <div style={{ marginLeft: 50 }}>
                    <Text>
                      The case has received area housing managerâ€™s final decision.
                    </Text>
                    <Field
                      id="decision"
                      name="decision"
                      label="The decision is"
                      type="radio"
                    >
                      <RadioGroup>
                        <Radio id="ho-review" value={Decision.Approve}>
                          {locale.views.tenureInvestigation.approve}
                        </Radio>
                        <Radio id="ho-review" value={Decision.Decline}>
                          {locale.views.tenureInvestigation.decline}
                        </Radio>
                      </RadioGroup>
                    </Field>
                    <InlineField name="confirm">
                      <Checkbox id="confirm" name="confirm">
                        I confirm that this is an instruction received by the housing area
                        manager
                      </Checkbox>
                    </InlineField>
                    <InlineField name="managerName">
                      <Input
                        id="managerName"
                        style={{ marginLeft: 53, marginTop: 22, maxWidth: 300 }}
                        placeholder="Enter manager's name"
                      />
                    </InlineField>
                  </div>
                )}
              </RadioGroup>
            </Field>

            <Button type="submit" disabled={disabled} style={{ width: 222 }}>
              {locale.confirm}
            </Button>
          </Form>
        )}
      </Formik>
    </>
  );
};
