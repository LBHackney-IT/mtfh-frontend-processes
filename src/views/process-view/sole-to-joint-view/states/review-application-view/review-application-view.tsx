import React, { useState } from "react";

import { CloseProcessView } from "../../../../../components";
import { locale } from "../../../../../services";
import { IProcess } from "../../../../../types";
import { isCurrentState } from "../../../../../utils/processUtil";
import { DesBox } from "../../../shared/process-components";
import { SubmitCaseView } from "../../../shared/submit-case-view";
import { HoReviewFailedView } from "../ho-review-view/ho-review-failed-view";
import { HoReviewView } from "../ho-review-view/ho-review-view";
import { NewTenancyView } from "../new-tenancy-view/new-tenancy-view";
import {
  EligibilityChecksPassedBox,
  TenureInvestigationRecommendationBox,
} from "../shared";
import { TenureInvestigationView } from "../tenure-investigation-view";

import { Process } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Box,
  Center,
  ErrorSummary,
  Spinner,
  StatusErrorSummary,
  StatusHeading,
} from "@mtfh/common/lib/components";

const { views } = locale;

interface ReviewApplicationViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
}

export const ReviewApplicationView = ({
  processConfig,
  process,
  mutate,
  optional,
}: ReviewApplicationViewProps): JSX.Element => {
  const [globalError, setGlobalError] = useState<number>();
  const [documentsSigned, setDocumentsSigned] = useState<boolean>(false);
  const { submitted, closeCase, setCloseCase } = optional;
  const { data: tenure, error } = useTenure(process.targetId);
  const {
    applicationSubmitted,
    tenureInvestigationFailed,
    tenureInvestigationPassed,
    tenureInvestigationPassedWithInt,
    hoApprovalPassed,
    hoApprovalFailed,
    tenureAppointmentScheduled,
    tenureAppointmentRescheduled,
    interviewScheduled,
    interviewRescheduled,
    tenureUpdated,
    processClosed,
  } = processConfig.states;

  if (error) {
    return (
      <ErrorSummary
        id="tenure-investigation-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!tenure) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const tenant = tenure?.householdMembers.find((m) => m.isResponsible);

  if (applicationSubmitted.state === process.currentState.state && submitted) {
    return (
      <SubmitCaseView
        processConfig={processConfig}
        process={process}
        mutate={mutate}
        optional={optional}
      />
    );
  }

  return (
    <div data-testid="soletojoint-ReviewApplication">
      {globalError && (
        <StatusErrorSummary id="review-application-global-error" code={globalError} />
      )}
      <EligibilityChecksPassedBox />
      <DesBox
        title={views.submitCase.supportingDocumentsApproved}
        description={views.submitCase.viewDocumentsOnDes}
      />

      {!isCurrentState(applicationSubmitted.state, process) && (
        <TenureInvestigationRecommendationBox
          processConfig={processConfig}
          process={process}
        />
      )}

      {applicationSubmitted.state === process.currentState.state && (
        <TenureInvestigationView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {[
        tenureInvestigationFailed.state,
        tenureInvestigationPassed.state,
        tenureInvestigationPassedWithInt.state,
        interviewScheduled.state,
        interviewRescheduled.state,
      ].includes(process.currentState.state) && (
        <HoReviewView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{ tenant }}
        />
      )}

      {(isCurrentState(hoApprovalFailed.state, process) ||
        isCurrentState(processClosed.state, process)) && (
        <HoReviewFailedView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
        />
      )}

      {[
        hoApprovalPassed.state,
        tenureAppointmentScheduled.state,
        tenureAppointmentRescheduled.state,
        tenureUpdated.state,
      ].includes(process.currentState.state) && (
        <NewTenancyView
          processConfig={processConfig}
          process={process}
          mutate={mutate}
          setGlobalError={setGlobalError}
          optional={{
            closeCase,
            setCloseCase,
            documentsSigned,
            setDocumentsSigned,
            tenant,
          }}
        />
      )}

      {closeCase && (
        <>
          <Box variant="warning">
            <StatusHeading
              variant="warning"
              title={views.closeProcess.soleToJointClosed}
            />
          </Box>
          <CloseProcessView
            process={process}
            processConfig={processConfig}
            mutate={mutate}
            setGlobalError={setGlobalError}
          />
        </>
      )}
    </div>
  );
};
