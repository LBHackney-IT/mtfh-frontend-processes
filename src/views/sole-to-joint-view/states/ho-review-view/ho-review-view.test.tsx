import React from "react";

import {
  getContactDetailsV2,
  mockContactDetailsV2,
  mockProcessV1,
  patchProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen, waitForElementToBeRemoved } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale, processes } from "../../../../services";
import { typeDateTime } from "../../../../test-utils";
import { ReviewApplicationView } from "../review-application-view/review-application-view";
import { HoReviewView } from "./ho-review-view";

import * as processV1 from "@mtfh/common/lib/api/process/v1/service";
import * as errorMessages from "@mtfh/common/lib/hooks/use-error-codes";

let submitted = false;
let closeCase = false;
const setSubmitted = () => {};
const setCloseCase = () => {};

describe("ho-review-view", () => {
  beforeEach(() => {
    jest.resetModules();
    submitted = false;
    closeCase = false;
    jest.spyOn(errorMessages, "useErrorCodes").mockReturnValue({});
  });

  test("it renders ReviewApplication view correctly for HOApprovalPassed state", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    const { container } = render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "HOApprovalPassed" },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await expect(
      screen.findByText(locale.views.tenureInvestigation.mustMakeAppointment, {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText("Date")).resolves.toBeInTheDocument();
    await expect(screen.findByText("Time")).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("it renders ReviewApplication view correctly for TenureInvestigationPassed state", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureInvestigationPassed",
          },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await expect(
      screen.findByText("approve application", { exact: false }),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ReviewApplication view correctly for TenureInvestigationFailed state", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureInvestigationFailed",
          },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await expect(
      screen.findByText("decline application", { exact: false }),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ReviewApplication view correctly for tenureInvestigationPassedWithInt state", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    render(
      <ReviewApplicationView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "tenureInvestigationPassedWithInt",
          },
        }}
        mutate={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await expect(
      screen.findByText("Interview Applicant", { exact: false }),
    ).resolves.toBeInTheDocument();
  });

  test("it shows/hides fields on radio button clicks", async () => {
    server.use(getContactDetailsV2(mockContactDetailsV2));
    render(
      <HoReviewView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureInvestigationFailed",
          },
        }}
        mutate={() => {}}
        setGlobalError={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    const appointmentRadio = screen.getByLabelText(
      locale.views.hoReviewView.makeAppointment,
      {
        exact: false,
      },
    );
    const reviewRadio = screen.getByText(locale.views.hoReviewView.passedForReview, {
      exact: false,
    });
    await expect(
      screen.queryByText(locale.views.hoReviewView.appointmentContactDetails),
    ).toBeNull();
    await userEvent.click(appointmentRadio);
    await expect(
      screen.findByText(locale.views.hoReviewView.appointmentContactDetails, {
        exact: false,
      }),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.queryByText(locale.views.hoReviewView.receivedDecision),
    ).toBeNull();
    await userEvent.click(reviewRadio);
    await expect(
      screen.findByText(locale.views.hoReviewView.receivedDecision),
    ).resolves.toBeInTheDocument();
    await userEvent.click(screen.getByText(locale.views.tenureInvestigation.approve));
  });

  test("it submits review", async () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(() => [false, setState]);
    const editProcessSpy = jest.spyOn(processV1, "editProcess");
    server.use(getContactDetailsV2(mockContactDetailsV2));
    server.use(patchProcessV1({}, 200));
    render(
      <HoReviewView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureInvestigationFailed",
          },
        }}
        mutate={() => {}}
        setGlobalError={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await userEvent.click(
      screen.getByLabelText(locale.views.hoReviewView.passedForReview, {
        exact: false,
      }),
    );
    await userEvent.click(
      screen.getByLabelText(locale.views.tenureInvestigation.approve),
    );
    await userEvent.type(
      screen.getByPlaceholderText(locale.views.hoReviewView.managersName),
      "test",
    );
    await userEvent.click(
      screen.getByText(locale.views.hoReviewView.confirmInstructionReceived),
    );
    await userEvent.click(screen.getByText(locale.confirm));
    expect(editProcessSpy).toHaveBeenCalledWith(
      expect.objectContaining({
        formData: { hoRecommendation: "approve", housingAreaManagerName: "test" },
        processTrigger: "HOApproval",
      }),
    );
  });

  test("it submits appointment", async () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(() => [false, setState]);
    const editProcessSpy = jest.spyOn(processV1, "editProcess");
    server.use(getContactDetailsV2(mockContactDetailsV2));
    server.use(patchProcessV1({}, 200));
    render(
      <HoReviewView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "TenureInvestigationFailed",
          },
        }}
        mutate={() => {}}
        setGlobalError={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await userEvent.click(
      screen.getByLabelText(locale.views.hoReviewView.makeAppointment, {
        exact: false,
      }),
    );
    await typeDateTime(screen, userEvent, "2099");
    await userEvent.click(screen.getByText(locale.confirm));
    expect(editProcessSpy).toHaveBeenCalledWith(
      expect.objectContaining({
        processTrigger: "ScheduleInterview",
      }),
    );
  });

  test("it submits reschedule appointment", async () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(() => [false, setState]);
    const editProcessSpy = jest.spyOn(processV1, "editProcess");
    server.use(getContactDetailsV2(mockContactDetailsV2));
    server.use(patchProcessV1({}, 200));
    render(
      <HoReviewView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "InterviewScheduled",
            processData: {
              formData: {
                appointmentDateTime: "2010-10-12T08:59:00.000Z",
              },
              documents: [],
            },
          },
        }}
        mutate={() => {}}
        setGlobalError={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await userEvent.click(
      screen.getByLabelText(locale.views.hoReviewView.makeAppointment, {
        exact: false,
      }),
    );
    await typeDateTime(screen, userEvent, "2099");
    await userEvent.click(screen.getByText(locale.confirm));
    expect(editProcessSpy).toHaveBeenCalledWith(
      expect.objectContaining({
        processTrigger: "RescheduleInterview",
      }),
    );
  });

  test("it submits schedule appointment date not passed", async () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, "useState");
    useStateSpy.mockImplementation(() => [false, setState]);
    const editProcessSpy = jest.spyOn(processV1, "editProcess");
    server.use(getContactDetailsV2(mockContactDetailsV2));
    server.use(patchProcessV1({}, 200));
    render(
      <HoReviewView
        processConfig={processes.soletojoint}
        process={{
          ...mockProcessV1,
          currentState: {
            ...mockProcessV1.currentState,
            state: "InterviewScheduled",
            processData: {
              formData: {
                appointmentDateTime: "2099-10-12T08:59:00.000Z",
              },
              documents: [],
            },
          },
        }}
        mutate={() => {}}
        setGlobalError={() => {}}
        optional={{ submitted, setSubmitted, closeCase, setCloseCase }}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await userEvent.click(
      screen.getByLabelText(locale.views.hoReviewView.makeAppointment, {
        exact: false,
      }),
    );
    await typeDateTime(screen, userEvent, "2099");
    await userEvent.click(screen.getByText(locale.confirm));
    expect(editProcessSpy).toHaveBeenCalledWith(
      expect.objectContaining({
        processTrigger: "",
      }),
    );
  });
});
