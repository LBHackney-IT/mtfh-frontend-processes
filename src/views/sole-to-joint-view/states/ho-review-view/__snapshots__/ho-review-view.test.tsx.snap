// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ho-review-view it renders ReviewApplication view correctly for HOApprovalPassed state 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewApplication"
  >
    <h1
      class="lbh-heading-h1"
    >
      Sole tenant requests a joint tenure
    </h1>
    <h2
      class="lbh-heading-h2 entity-summary__tenure-heading"
    >
      <div>
        Tenure payment ref
         
        mt0xzvjp6g
      </div>
      <div>
        72251 Champlin Trafficway, 99647
      </div>
      <div>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/person/a302fe28-e5c3-4c84-a3e6-cd575e65093f"
          target="_blank"
        >
          Global Web Consultant Alverta Denesik
        </a>
      </div>
    </h2>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Supporting documents approved
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="http://des/teams/1/dashboard/requests"
          target="_blank"
        >
          View documents on the Document Evidence Store
        </a>
      </div>
    </div>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Tenure investigator recommendation: Interview Applicant
        </h4>
      </div>
    </div>
    <p
      class="lbh-body-s"
    >
      Global Web Consultant Alverta Denesik
       contact details:
    </p>
    <p
      class="lbh-body-s"
    >
      Phone:
      <span
        style="margin-left: 1em;"
      >
        075949281
      </span>
    </p>
    <p
      class="lbh-body-s"
    >
      Email:
      <span
        style="margin-left: 1em;"
      >
        Gay.Jacobi@hotmail.com
      </span>
    </p>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Sole to joint tenure application approved
        </h4>
      </div>
    </div>
    <h2
      class="lbh-heading-h2"
    >
      Sole to Joint application approved, next steps:
    </h2>
    <p
      class="lbh-body-m"
    >
      You must make an office appointment with the applicant to sign a new tenancy agreement
    </p>
    <form
      action="#"
      class="request-appointment-form"
      id="request-appointment-form"
      novalidate=""
    >
      <div
        style="display: flex;"
      >
        <fieldset
          class="govuk-form-group lbh-form-group mtfh-appointment-form__date"
          id="appointment-form-date"
          style="margin-top: 1.5em; width: 100%;"
        >
          <legend
            class="govuk-label lbh-label"
            for="appointment-form-date-field"
          >
            Date
            <sup
              aria-hidden="true"
            >
              *
            </sup>
          </legend>
          <div
            class="govuk-date-input lbh-date-input"
          >
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-date-field-day"
              />
              <input
                aria-label="Day"
                aria-valuemax="31"
                aria-valuemin="1"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="day"
                placeholder="dd"
                required=""
                role="spinbutton"
                value=""
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-date-field-month"
              />
              <input
                aria-label="Month"
                aria-valuemax="12"
                aria-valuemin="1"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="month"
                placeholder="mm"
                required=""
                role="spinbutton"
                value=""
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-date-field-year"
              />
              <input
                aria-label="Year"
                class="govuk-input lbh-input govuk-input govuk-date-input__input govuk-input--width-4"
                name="year"
                placeholder="yy"
                required=""
                role="spinbutton"
                value=""
              />
            </div>
          </div>
        </fieldset>
        <fieldset
          class="govuk-form-group lbh-form-group mtfh-appointment-form__time"
          id="appointment-form-time"
          style="margin-top: 1.5em; width: 100%;"
        >
          <legend
            class="govuk-label lbh-label"
            for="appointment-form-time-field"
          >
            Time
            <sup
              aria-hidden="true"
            >
              *
            </sup>
          </legend>
          <div
            class="govuk-date-input lbh-date-input"
          >
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-time-field-hour"
              />
              <input
                aria-label="Hour"
                aria-valuemax="12"
                aria-valuemin="0"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="hour"
                placeholder="00"
                required=""
                role="spinbutton"
                value=""
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-time-field-minute"
              />
              <input
                aria-label="Minute"
                aria-valuemax="59"
                aria-valuemin="0"
                class="govuk-input lbh-input govuk-date-input__input govuk-input--width-2"
                name="minute"
                placeholder="00"
                required=""
                role="spinbutton"
                value=""
              />
            </div>
            <div
              class="govuk-date-input__item"
            >
              <label
                class="govuk-label lbh-label"
                for="appointment-form-time-field-amPm"
              />
              <select
                aria-label="AM/PM"
                class="govuk-select lbh-select"
                id="amPm"
                name="amPm"
              >
                <option
                  value=""
                >
                  AM/PM
                </option>
                <option
                  value="am"
                >
                  AM
                </option>
                <option
                  value="pm"
                >
                  PM
                </option>
              </select>
            </div>
          </div>
        </fieldset>
      </div>
      <button
        class="govuk-button lbh-button"
        disabled=""
        style="width: 222px;"
        type="submit"
      >
        Continue
      </button>
    </form>
  </div>
</div>
`;
